### ISSUE TITLE
Doesn't honor tsconfig "target" property

### ISSUE THREAD

**Issue by nibble-4bits** [huozhi/bunchee#153]
Hi @huozhi,

I have set up bunchee according to the README, to bundle a TypeScript library that I'm working on. When I run it, the bundle is created without issues, however, it seems bunchee is not respecting the `"target"` property in tsconfig.json and instead defaults to using `es5` as bundling target.

This is my tsconfig.json:

```json
{
  "extends": "@tsconfig/node16-strictest/tsconfig.json",
  "compilerOptions": {
    "outDir": "build",
    "target": "ES6",
    "declaration": true,
    "noUncheckedIndexedAccess": false
  },
  "include": ["src/**/*.ts"],
  "exclude": ["node_modules"]
}
```

This is my bunchee config in the package.json:

```
{
  "type": "module",
  "main": "build/main.cjs",
  "types": "build/main.d.ts",
  "exports": {
    "require": "./build/main.cjs",
    "import": "./build/main.esm.js",
    "default": "./build/main.esm.js"
  },
  "scripts": {
    "build": "bunchee src/main.ts",
  },
  "devDependencies": {
    "bunchee": "^2.1.7",
  },
}
```

The code generated in the `build/` directory contains some autogenerated polyfills which are not needed for ES6 code, which is an indication that the bundling is targeting ES5 instead of the ES6 target specified in the tsconfig.json. If I change the `build` script in package.json to `bunchee --target es2015 src/main.ts`, the generated bundle is different and doesn't contain all of those ES5 polyfills.

**Comment by huozhi** [comment]
Yeah I have an issue for that here https://github.com/huozhi/bunchee/issues/146

**Comment by huozhi** [comment]
Landed in [v2.2.0](https://github.com/huozhi/bunchee/releases/tag/v2.2.0) 🚀
